### PR TITLE
feat: Make onError middleware option affect middleware behaviour, exp…

### PR DESCRIPTION
**Pull request by yourtallness**
…ose a strict middleware variant for parity with the Next case

### CODE REVIEW

File: src/Clerk.ts [hunk 1]
```diff
@@ -133,7 +133,22 @@ export default class Clerk {
...
 
-  expressMiddleware({ onError }: MiddlewareOptions = {}) {
+  // defaultOnError doesn't do anything
+  // return an Error with statusCode if you want to halt the request chain
+  defaultOnError(error: Error) {
+    console.error(error.message);
+  }
+
+  // strictOnError returns an error with statusCode 401 so that express will halt the request chain
+  strictOnError(error: Error) {
```
Comment (colinclerk) [inline src/Clerk.ts:143]: Will this replace 5xx errors from clerk with a 401 back to the browser?

Comment (yourtallness) [reply src/Clerk.ts:143]: Yes, it currently doesn't discern between error cases.

File: src/Clerk.ts [hunk 2]
```diff
@@ -133,7 +133,22 @@ export default class Clerk {
...
+    console.error(error.message);
+    const err = new Error("Unauthenticated Access");
+    // @ts-ignore
```
Comment (SokratisVidros) [inline src/Clerk.ts:146]: ❓ Can we avoid this @ts-ignore by defining the error type?

**Review by SokratisVidros — APPROVED** [review]

File: src/Clerk.ts (hunk 2 above)
Comment (yourtallness) [reply src/Clerk.ts:146]: Will do

File: src/Clerk.ts (hunk 1 above)
Comment (yourtallness) [reply src/Clerk.ts:143]: Think I'm going to define a few error types to tell these cases apart:

- No cookie
- No session
- Miscellaneous error

Will need to see how to bring the same functionality to the next middleware somehow.

File: src/utils/Errors.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export class HttpError extends Error {
+    public statusCode: number;
+    public body: unknown;
+
+    constructor(statusCode: number, message: string, body: unknown) {
+        super(message);
+        this.statusCode = statusCode;
+        this.body = body;
```
Comment (SokratisVidros) [inline src/utils/Errors.ts:8]: 🙃 

```suggestion
        this.data = data;
```

**Review by SokratisVidros — APPROVED** [review]

File: src/utils/Errors.ts (hunk 3 above)
Comment (yourtallness) [reply src/utils/Errors.ts]: Done

File: src/Clerk.ts (hunk 1 above)
Comment (yourtallness) [reply src/Clerk.ts:144]: Implemented pass through of server errors

**Pull request merged**